cdap-security/src/main/java/io/cdap/cdap/security/store/FileSecureStoreImporter.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.store;
+
+import io.cdap.cdap.api.security.store.SecureStoreData;
+import io.cdap.cdap.security.store.file.FileSecureStoreCodec;
+import io.cdap.cdap.security.store.file.KeyInfo;
+import io.cdap.cdap.security.store.file.SecureStoreDataCodecV1;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Enumeration;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+/**
+ * Service responsible for importing/upgrading the {@link java.security.KeyStore} files saved by the
+ * {@link FileSecureStoreService}.
+ */
+public class FileSecureStoreImporter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(FileSecureStoreImporter.class);
+  private static final Class<? extends FileSecureStoreCodec>[] HISTORICAL_CODECS = new Class[]{
+    SecureStoreDataCodecV1.class
+  };
+
+  private final FileSecureStoreCodec currentCodec;
+
+  public FileSecureStoreImporter(FileSecureStoreCodec currentCodec) {
+    this.currentCodec = currentCodec;
+  }
+
+  /**
+   * Attempts to import a prior KeyStore file from the specified path.
+   * @param keyStore The keystore to import into
+   * @param path The filepath of the prior keystore
+   * @param password The password to use
+   * @throws IOException If the import failed
+   * @throws IllegalArgumentException If an invalid argument was specified
+   */
+  public void importFromPath(KeyStore keyStore, Path path, char[] password) throws IOException,
+    IllegalArgumentException {
+    try {
+      try (InputStream in = new DataInputStream(Files.newInputStream(path))) {
```

Comment:
Seems like these two try(s) can be combined to one?

Reply:
Fixed